### PR TITLE
Add new case to test if cpu hotplug causes time jump

### PR DESCRIPTION
**Pull request by huangyum**
Signed-off-by: Yumei Huang <yuhuang@redhat.com>
ID: 1947341

### CODE REVIEW

**Comment by huangyum** [comment]
@yanan-fu @PaulYuuu  Pls help review, thanks. 
(1/5) repeat1.Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.cpu_device_hotplug_time_jump: PASS (10883.91 s)
 (2/5) repeat2.Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.cpu_device_hotplug_time_jump: PASS (10881.53 s)
 (3/5) repeat3.Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.cpu_device_hotplug_time_jump: PASS (10880.91 s)
 (4/5) repeat4.Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.cpu_device_hotplug_time_jump: PASS (10881.90 s)
 (5/5) repeat5.Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.cpu_device_hotplug_time_jump: PASS (10940.20 s)

File: qemu/tests/cpu_device_hotplug_time_jump.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    error_context.context("Check number of CPU inside guest.", logging.info)
+    if not cpu_utils.check_if_vm_vcpus_match_qemu(vm):
+        test.fail("Actual number of guest CPUs is not equal to expected")
+
+    error_context.context("Check if guest has time jump", logging.info)
+    output = session.cmd_output("dmesg")
+    session.close()
+    time1 = float(re.findall(r"\[\s*(\d+\.?\d+)\]\s+CPU.*has been hot-added",
+                             output)[0])
```
Comment (yanan-fu) [inline qemu/tests/cpu_device_hotplug_time_jump.py:44]: How about `re.findall(r"^\[\s*(\d+\.?\d+)\]\s+CPU.*has been hot-added$", content, re.M)[0]` ?

File: qemu/tests/cpu_device_hotplug_time_jump.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    time2 = float(re.findall(r"\[\s*(\d+\.?\d+)\]\s+Will online and init "
+                             "hotplugged CPU", output)[0])
+    time_gap = time2 - time1
+    logging.info("The time gap is %.6fs", time_gap)
+    expected_gap = float(params["expected_gap"])
```
Comment (PaulYuuu) [inline qemu/tests/cpu_device_hotplug_time_jump.py:49]: `params.get_numeric("expected_gap", float)`

**Review by PaulYuuu — COMMENTED** [review]
The code looks good to me overall, but I am not sure if s390x needs it.

@fbq815 Are you interested in testing it on 390x?

File: qemu/tests/cpu_device_hotplug_time_jump.py (hunk 1 above)
Comment (huangyum) [reply qemu/tests/cpu_device_hotplug_time_jump.py:44]: Would you pls explain why? What the difference besides the '$' in the str end?

Comment (yanan-fu) [reply qemu/tests/cpu_device_hotplug_time_jump.py:44]: IMO, Multi-line match is more accurate than match with the whole content in this situation.  The key point is about the `$`, but `^`, `$` and `re.M` work together.

Comment (huangyum) [reply qemu/tests/cpu_device_hotplug_time_jump.py]: It looks it works same in this case. The keywords are accurate enough to find the first matching content. Anyway, I updated as you suggested. Please review again. Thanks.

**Comment by yanan-fu** [comment]
> The code looks good to me overall, but I am not sure if s390x needs it.
> 
> @fbq815 Are you interested in testing it on 390x?

@fbq815  Could you check this ? Does s390x need this test scenario? Thanks!

**Comment by fbq815** [comment]
> > The code looks good to me overall, but I am not sure if s390x needs it.
> > @fbq815 Are you interested in testing it on 390x?
> 
> @fbq815 Could you check this ? Does s390x need this test scenario? Thanks!

working on this now will give the answer soon

**Comment by fbq815** [comment]
> > > The code looks good to me overall, but I am not sure if s390x needs it.
> > > @fbq815 Are you interested in testing it on 390x?
> > 
> > 
> > @fbq815 Could you check this ? Does s390x need this test scenario? Thanks!
> 
> working on this now will give the answer soon

Since s390x only support with hotplug-cpu, I think we could not add this case until now

**Comment by yanan-fu** [comment]
Thanks all!    Ack

**Pull request merged**